hydroshare_jupyter_sync/lib/resource_strategies.py

```diff
@@ -0,0 +1,57 @@
+from abc import ABC, abstractmethod
+import shutil
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from hsclient import Resource
+from zipfile import ZipFile
+
+
+class AbstractHydroShareEntityDownloadStrategy(ABC):
+    def __init__(self, resource: Resource, data_path: str):
+        self.resource = resource
+        self.data_path = data_path
+
+    @abstractmethod
+    def download(self, path: str):
+        """Interface for HydroShare file system entity download"""
+
+    def create_intermediary_directories(self, path: Path) -> Path:
+        """Handles the creation of {data_path}/{resource_id}/data/contents"""
+        contents_path = (
+            Path(self.data_path) / self.resource.resource_id / "data/contents" / path
```

Comment:
`"data/contents"` isn't windows safe

Reply:
or maybe it is since it's passed to Path?  I'm not sure now, but it is incosistent.